tf-psa-crypto/drivers/builtin/src/psa_crypto_ecp.c

```diff
@@ -617,6 +617,35 @@ psa_status_t mbedtls_psa_generate_key_iop_setup(
     return mbedtls_to_psa_error(status);
 }
 
+psa_status_t mbedtls_psa_generate_key_iop_complete(
```

Comment:
AFAIK we are not going to implement FFDH either at least in the scope of the current work (https://github.com/Mbed-TLS/mbedtls/pull/9490#discussion_r1734837190), so if the only type of key generation we are going to have is ECP there is no much point of prefixing it with `mbedtls_psa_ecp` and then have another function that only forward parameters to it, it might make it easier later if we decide to add ffdh_iop_generate_key() but from where I see it it's out of scope for current work.

Reply:
I think that the `mbedtls_psa_ecp` prefix is more like a convention used throughout this file and generally the `mbedtls_psa_<module name>` pattern throughout the library. The non-interruptible version too is called `mbedtls_psa_ecp_generate_key`. We shouldn't break this convention without a good justification.

Looking at this made me realise that I missed another problem. The function declarations corresponding to functions in `tf-psa-crypto/drivers/builtin/src/psa_crypto_ecp.c` should be in`tf-psa-crypto/drivers/builtin/src/psa_crypto_ecp.h`. As far as I can tell, this PR puts them in `tf-psa-crypto/core/psa_crypto_core.h`.

Reply:
We're even less likely to do interruptible FFDH than RSA. Interruptible operations are for platforms that are very resource-constrained and they will use the fastest algorithm that's secure enough. FFDH is a lot slower than ECDH for the same security level, and even non-resource constrained platforms rarely use FFDH these days.

On the other hand, we may plausibly implement interruptible operations for post-quantum algorithms at some point in a few years. So the level of indirection is meaningful, but it won't be useful for several years.

From a code maintenance perspective, however, I agree with Valerio that it's weird to have an exported function in `psa_crypto_ecp.c` that doesn't have `ec` in its name. That's likely to make things confusing when debugging.